### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by julshvets**
[DEMO LINK](https://julshvets.github.io/react_goods-selector/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -15,71 +19,91 @@ export const goods = [
...
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+  handleClickClearButton = () => {
```
Comment (naziamoff) [inline src/App.tsx:27]: too wordy. You can have it concise without the loss of meaning
```suggestion
  clearSelection = () => {
```

File: src/App.tsx [hunk 2]
```diff
@@ -15,71 +19,91 @@ export const goods = [
...
-            Dumplings
-          </td>
-        </tr>
+  handleClickSelectedButton = (good:string) => {
+    if (this.state.selectedGood === good) {
+      this.setState({ selectedGood: '' });
+    } else {
+      this.setState({ selectedGood: good });
+    }
+  };
```
Comment (naziamoff) [inline src/App.tsx:37]: you already have reset method, so let's follow DRY

File: src/App.tsx [hunk 3]
```diff
@@ -15,71 +19,91 @@ export const goods = [
...
+                  data-cy="ClearButton"
+                  type="button"
+                  className="delete ml-3"
+                  onClick={this.handleClickClearButton}
+                />
+              </>
+            )
+            : (
+              <>No goods selected</>
+            )}
```
Comment (naziamoff) [inline src/App.tsx:60]: seems the fragment is redudant
```suggestion
            : No goods selected}
```

File: src/App.tsx [hunk 4]
```diff
@@ -15,71 +19,91 @@ export const goods = [
...
-          </td>
+        <table className="table">
+          <tbody>
+            {goods.map(good => (
+              <tr
+                key={good}
+                data-cy="Good"
+                className={selectedGood === good
+                  ? 'has-background-success-light'
+                  : ''}
```
Comment (naziamoff) [inline src/App.tsx:71]: [classnames](https://www.npmjs.com/package/classnames)

**Review by naziamoff — CHANGES_REQUESTED** [review]
not bad, a couple of fixes and we're good to go

File: src/App.tsx (hunk 4 above)
Comment (julshvets) [reply src/App.tsx]: Thank you!

File: src/App.tsx [hunk 5]
```diff
@@ -15,71 +20,85 @@ export const goods = [
...
-          </td>
+  clearSelection = () => {
+    this.setState({ selectedGood: '' });
+  };
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
+  setSelection = (good:string) => {
```
Comment (naziamoff) [inline src/App.tsx:32]: let's be more specific in our naming
```suggestion
  setSelectedGood = (good:string) => {
```

File: src/App.tsx [hunk 6]
```diff
@@ -15,71 +20,85 @@ export const goods = [
...
+                })}
+              >
+                <td>
+                  {selectedGood === good
+                    ? (
+                      <button
+                        type="button"
+                        className="button is-info"
+                        onClick={(
+                          () => this.clearSelection())}
```
Comment (naziamoff) [inline src/App.tsx:75]:
```suggestion
                        onClick={this.clearSelection}
```
